pkg/sources/sources.go

```diff
@@ -44,6 +44,19 @@ type Source interface {
 	GetProgress() *Progress
 }
 
+// SourceUnitUnmarshaller defines an optional interface a Source can implement
+// to support units coming from an external source.
+type SourceUnitUnmarshaller interface {
+	UnmarshalSourceUnit(data []byte) (SourceUnit, error)
```

Comment:
Yes, most likely they'll be provided to the caller to be consumed. Marshalling can happen without the `Source` doing it for us.

It's an extra step the engine will have to take to scan units marshalled and stored in an external source.

Reply:
per huddle: they won't be completely opaque, because the source unit manager will be recording each source unit's id and using it to plan/report on things